src/test/java/org/apache/maven/plugins/dependency/AbstractDependencyMojoTestCase.java

```diff
@@ -35,26 +35,49 @@
 import org.eclipse.aether.RepositorySystemSession;
 import org.eclipse.aether.repository.LocalRepository;
 import org.eclipse.aether.repository.LocalRepositoryManager;
+import org.junit.Before;
 import org.sonatype.plexus.build.incremental.DefaultBuildContext;
 
 public abstract class AbstractDependencyMojoTestCase extends AbstractMojoTestCase {
 
     protected File testDir;
-
     protected DependencyArtifactStubFactory stubFactory;
 
-    protected void setUp(String testDirectoryName, boolean createFiles) throws Exception {
-        setUp(testDirectoryName, createFiles, true);
+    @Before
+    public void setUp() throws Exception {
+        // Initialize mojo lookups as required by AbstractMojoTestCase
+        super.setUp();
+
+        // Create temporary test directory
+        testDir = Files.createTempDirectory("test-dependency").toFile();
+        testDir.deleteOnExit();
+
+        // Initialize stub factory with default settings
+        stubFactory = new DependencyArtifactStubFactory(testDir, true, true);
     }
 
-    protected void setUp(String testDirectoryName, boolean createFiles, boolean flattenedPath) throws Exception {
-        // required for mojo lookups to work
-        super.setUp();
+    /**
+     * Allows subclasses to customize the setup with specific test directory name and stub factory settings.
+     *
+     * @param testDirectoryName The name for the temporary test directory.
+     * @param createFiles Whether to create files in the stub factory.
+     * @param flattenedPath Whether to use flattened paths in the stub factory.
+     * @throws Exception If setup fails.
+     */
+    protected void customizeSetUp(String testDirectoryName, boolean createFiles, boolean flattenedPath)
```

Comment:
Is this meant to be invoked or overridden? If the former, then make it final. But consider calling this from setUp and letting subclasses override it instead.